api/src/main/java/org/apache/commons/rdf/api/RDFTerm.java

```diff
@@ -37,4 +55,46 @@
      */
     String ntriplesString();
 
+	/**
+	 * Check it this RDFTerm is equal to another RDFTerm.
+	 * <p>
+	 * If this object is an {@link IRI}, equality is checked using
```

Comment:
I don't think that would make sense in `RDFTerm` as each of them override `.equals()`? What would the `default` method do and when would it be called? 

Another problem is that you can't provide a `default` method for methods from `java.lang.Object`.